src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/dynamic_coefficient.jl

```diff
@@ -42,6 +42,54 @@ in the `x` and `y` directions).
 
 `DynamicCoefficient` is updated according to `schedule`, and `minimum_numerator` defines the minimum
 value that is acceptable in the denominator of the final calculation.
+
+Examples
+========
+
+```jldoctest
+julia> using Oceananigans
+
+julia> dynamic_coeff = DynamicCoefficient(averaging=(1, 2))
+DynamicCoefficient with
+├── averaging = (1, 2)
+├── schedule = IterationInterval(1, 0)
+└── minimum_numerator = 1.0e-32
+
+julia> dynamic_smagorinsky = Smagorinsky(coefficient=dynamic_coeff)
+Smagorinsky closure with
+├── coefficient = DynamicCoefficient(averaging = (1, 2), schedule = IterationInterval(1, 0))
+└── Pr = 1.0
+```
+
+The dynamic Smagorinsky that was just created above has its dynamic coefficient recalculated at every time
+step. While this will provide the highest level of accuracy, it will almost certainly be very slow
+given the high cost of calculating `DynamicCoefficient`s. Because of that, it is common in the
```

Comment:
```suggestion
The dynamic Smagorinsky above has its dynamic coefficient recalculated at every time
step. While this provides the highest level of accuracy, it also brings about slowdown
given the high cost of calculating `DynamicCoefficient`s. Because of this slowdown, it is common in the
```

Reply:
```suggestion
The dynamic Smagorinsky above has its dynamic coefficient recalculated at every time
step. While this provides the highest level of accuracy, it will almost certainly be very slow
given the high cost of calculating `DynamicCoefficient`s. Because of this slowdown, it is common in the
```

Reply:
Sounds good!

Reply:
Just a remark that I think its misleading to refer the difference between a stride of 1 as yielding the "highest accuracy". I think a stride of 1 is simply an implementation of the scheme as written out mathematically, so apart from the discretization, there is no approximation or notion of "accuracy". When we use a stride different from 1 (eg 2), only then is the notion of "accuracy" introduced.

I think this can be clarified through a simplification of the text, something like: "to alleviate the high computational cost of the dynamic coefficient calculation, users may introduce an approximation wherein the dynamic coefficient is recomputed only every so often. This approximation is valid when the coefficient changes relatively slowly compared to a single time-step."

Reply:
Reworded everything based on @glwagner comments.